lib/OpenQA/WebAPI/Controller/Admin/User.pm

```diff
@@ -27,17 +27,25 @@ sub update {
         $is_operator = 1;
     }
 
+    my $err = 0;
+    my $msg = '';
+
     my $user = $set->find($self->param('userid'));
     if (!$user) {
-        $self->flash('error', "Can't find that user");
+        $err = 1;
+        $msg = "Can't find that user";
     }
     else {
         $user->update({is_admin => $is_admin, is_operator => $is_operator});
-        $self->flash('info', 'User ' . $user->nickname . ' updated');
+        $msg = 'User ' . $user->nickname . ' updated';
         $self->emit_event('user_update_res', {nickname => $user->nickname, role => $role});
     }
 
-    $self->redirect_to($self->url_for('admin_users'));
+    if ($self->tx->req->headers->header('Accept') eq 'application/json') {
```

Comment:
`Use of uninitialized value in string eq at /home/squamata/project/lib/OpenQA/WebAPI/Controller/Admin/User.pm line 44.` in t/ui/09-users-list.t

Reply:
```suggestion
    if (($self->tx->req->headers->accept // '') eq 'application/json') {
```